service-coordtransform/src/main/java/fi/nls/paikkatietoikkuna/coordtransform/CoordinateTransformationActionHandler.java

```diff
@@ -50,30 +107,87 @@ public void init() {
         if (endPoint == null) {
             endPoint = PropertyUtil.getNecessary(PROP_END_POINT);
         }
+        lineSeparators.put("win","\r\n");
+        lineSeparators.put("mac","\n");
+        lineSeparators.put("unix","\r");
+
+        coordinateSeparators.put("space", " ");
+        coordinateSeparators.put("tab", "\t");
+        coordinateSeparators.put("comma", ",");
     }
 
     @Override
     public void handleAction(ActionParameters params) throws ActionException {
+        hasMoreCoordinates = false;
         String sourceCrs = getSourceCrs(params);
         String targetCrs = getTargetCrs(params);
-        int dimension = sourceCrs.indexOf(',') > 0 ? 3 : 2;
-
-        List<Coordinate> coords;
-        try (InputStream in = params.getRequest().getInputStream()) {
-            coords = parseInputCoordinates(in, dimension);
-        } catch (IOException e) {
-            throw new ActionException("Failed to parse input JSON!");
+        String transformType = params.getHttpParam(PARAM_TRANSFORM_TYPE);
+        int sourceDimension = params.getRequiredParamInt(PARAM_SOURCE_DIMENSION);
+        int targetDimension = params.getRequiredParamInt(PARAM_TARGET_DIMENSION);
+        boolean transformToFile = false;
+        int queryDimension = sourceDimension;
+        boolean addZeroes = false;
+        if (sourceDimension == 2 && targetDimension == 3){
+            addZeroes = true;
+            queryDimension = 3; //parse added zeroes to query
+            sourceCrs = sourceCrs + ",EPSG:3900"; //add N2000 that coordtrans service doesn't fail
```

Comment:
Is there a requirement for 2D -> 3D to work?

Reply:
There is a use case where 2D coordinates are wanted to transform to 3D coordinate system to get transformed x and y coordinates. Maybe geocentric XYZ.

Reply:
OK